src/engine/fileOpsIns.cpp

```diff
@@ -470,77 +478,97 @@ void DivEngine::loadSBI(SafeReader& reader, std::vector<DivInstrument*>& ret, St
 
     int sbi_header = reader.readI();
     // SBI header determines format
-    bool is_2op = (sbi_header == 0x1A494253); // SBI\x1A
+    bool is_2op = (sbi_header == 0x1A494253 || sbi_header == 0x1A504F32); // SBI\x1A or 2OP\x1A
     bool is_4op = (sbi_header == 0x1A504F34); // 4OP\x1A
     bool is_6op = (sbi_header == 0x1A504F36); // 6OP\x1A - Freq Monster 801-specific
 
     // 32-byte null terminated instrument name
-    ins->name = reader.readString(32);
-
-    // 2op SBI
-    uint8_t sbi_Mcharacteristics = reader.readC();
-    uint8_t sbi_Ccharacteristics = reader.readC();
-    uint8_t sbi_Mscaling_output = reader.readC();
-    uint8_t sbi_Cscaling_output = reader.readC();
-    uint8_t sbi_Meg_AD = reader.readC();
-    uint8_t sbi_Ceg_AD = reader.readC();
-    uint8_t sbi_Meg_SR = reader.readC();
-    uint8_t sbi_Ceg_SR = reader.readC();
-    uint8_t sbi_Mwave = reader.readC();
-    uint8_t sbi_Cwave = reader.readC();
-    uint8_t sbi_FeedConnect = reader.readC();
-
-    // 4op SBI
-    uint8_t sbi_M4characteristics;
-    uint8_t sbi_C4characteristics;
-    uint8_t sbi_M4scaling_output;
-    uint8_t sbi_C4scaling_output;
-    uint8_t sbi_M4eg_AD;
-    uint8_t sbi_C4eg_AD;
-    uint8_t sbi_M4eg_SR;
-    uint8_t sbi_C4eg_SR;
-    uint8_t sbi_M4wave;
-    uint8_t sbi_C4wave;
-    uint8_t sbi_4opConnect;
+    String patchName = reader.readString(32);
+    patchName = (patchName.length() == 0) ? stripPath : patchName;
+
+    typedef struct {
```

Comment:
Hmmm... this is C++. `typedef` is not necessary.
You could also move this outside loadSBI.

Reply:
Done